gui/src/components/Taipy/TableFilter.tsx

```diff
@@ -0,0 +1,364 @@
+                        <CheckIcon />
+                    </IconButton>
+                </span>
+            </Tooltip>
+            <Tooltip title="Delete">
+                <span>
+                    <IconButton onClick={onDeleteClick} disabled={!enableDel}>
+                        <DeleteIcon />
+                    </IconButton>
+                </span>
+            </Tooltip>
+        </Box>
+    );
+};
+
+const TableFilter = (props: TableFilterProps) => {
+    const { onValidate } = props;
+
+    const [showFilter, setShowFilter] = useState(false);
+    const filterRef = useRef<HTMLButtonElement | null>(null);
+    const [filters, setFilters] = useState<Array<FilterDesc>>([]);
+
+    const onShowFilterClick = useCallback(() => setShowFilter((f) => !f), []);
+
+    const updateFilter = useCallback((idx: number, nfd: FilterDesc, remove?: boolean) => {
+        setFilters((fds) => {
+            if (idx > -1) {
+                if (remove) {
+                    fds.splice(idx, 1);
+                    return [...fds];
+                }
+                return fds.map((fd, index) => (index == idx ? nfd : fd));
+            }
+            if (remove) {
+                return fds;
+            }
+            return [...fds, nfd];
+        });
+    }, []);
+
+    const onApply = useCallback(() => {
+        onValidate(filters);
+        onShowFilterClick();
+    }, [onValidate, filters, onShowFilterClick]);
+    const onRemove = useCallback(() => {
+        onValidate([]);
+        onShowFilterClick();
+    }, [onValidate, onShowFilterClick]);
+
+    return (
+        <>
+            <Tooltip title="Filter list">
+                <IconButton onClick={onShowFilterClick} size="small" sx={iconInRowSx} ref={filterRef}>
+                    <FilterListIcon />
+                </IconButton>
+            </Tooltip>
+            <Popover
+                anchorEl={filterRef.current}
+                anchorOrigin={anchorOrigin}
+                open={showFilter}
+                onClose={onShowFilterClick}
+            >
+                <Box sx={filtersSx}>
+                    {filters.map((fd, idx) => (
+                        <FilterRow
+                            key={"fd" + idx}
+                            idx={idx}
+                            filter={fd}
+                            columns={props.columns}
+                            colsOrder={props.colsOrder}
+                            setFilter={updateFilter}
+                        />
+                    ))}
+                    <FilterRow
+                        idx={-(filters.length + 1)}
+                        columns={props.columns}
+                        colsOrder={props.colsOrder}
+                        setFilter={updateFilter}
+                    />
+                    <Box sx={buttonBoxSx}>
+                        <Button endIcon={<ClearIcon />} onClick={onRemove}>{`Reset list (remove applied filter${
```

Comment:
I couldn't spot the case where there are no filters yet.

Reply:
the aim of this button is to remove all the applied filters: same as deleting all filters and then clicking apply but without removing the filters from the list

Reply:
I'll check this out when it runs on my screen...